includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -59,7 +60,7 @@ class UPE_Payment_Gateway extends WC_Payment_Gateway_WCPay {
 	 *
 	 * @var UPE_Payment_Method
 	 */
-	protected $payment_method;
+	public $payment_method;
```

Comment:
This comment has nothing to do with this line, but do you think it might be useful to move all the AJAX actions in the constructor of this class to the new `WC_Payments_UPE_Checkout` class instead? This way we can ensure they only fire once, without having to check `has_action` before adding each hook.

Reply:
That'd be great! After a quick try, I can say that it doesn't work. We need to make this work, as the checkout class seems to be the ideal place for these hooks definitions.